src/App/App.js

```diff
@@ -34,61 +34,49 @@ class App extends Component {
     return (
       <div className='App'>
         <Header 
-        changeView={this.changeView}
         loggedIn={this.state.loggedIn} 
         updateLoginStatus={this.updateLoginStatus} 
         updateUserId={this.updateUserId}
         />
-        {this.state.view === 'homepage' && 
-          <>
-            <main>
-              {this.state.error && 
-              <h3 className='error-msg'>{this.state.error}</h3>
-              }
-              <Movies 
-              movies={this.state.movies} 
-              loggedIn={this.state.loggedIn}
-              userRatings={this.state.userRatings}
-              updateCurrentMovie={this.updateCurrentMovie}
-              />
-            </main>
-          </> 
-        }
-
-        {this.state.view === 'login' && 
+        <Route exact path='/' render={() => 
+          <Movies 
+            error={this.state.error}
+            movies={this.state.movies} 
+            loggedIn={this.state.loggedIn}
+            userRatings={this.state.userRatings}
+            updateCurrentMovie={this.updateCurrentMovie}
+          />} />
+        <Route exact path='/login' render={() => 
           <Login 
             updateUserId={this.updateUserId}
             updateLoginStatus={this.updateLoginStatus} 
-            changeView={this.changeView}
             error={this.state.error} 
+            loggedIn={this.state.loggedIn}
             updateError={this.updateError}
             getUserRatings={this.getUserRatings}
-          />
-        }
-
-        {this.state.view === 'movie-details' &&
-          <MovieDetails 
-            poster={this.state.currentMovie.poster_path}
-            title={this.state.currentMovie.title}
-            releaseDate={this.state.currentMovie.release_date}
-            averageRating={this.state.currentMovie.average_rating}
-            userRatings={this.state.userRatings}
-            currentMovie={this.state.currentMovie}
-            currentMovieRatingInfo={this.state.currentMovieRatingInfo}
-            loggedIn={this.state.loggedIn}
-            userId={this.state.userId}
-            updateUserRatings={this.updateUserRatings}
-            findCurrentMovieRating={this.findCurrentMovieRating}
-          />
-        }
+        />} />
+        <Route 
+          path='/movies/:id'
+          exact
+          render={({ match }) => {
+            const movieToRender = this.state.movies.find(movie => movie.id === +match.params.id)
+            return (
+              <MovieDetails 
```

Comment:
glad we were able to figure out how to pass the props using the spread operator here!